external/buildscripts/build_all_osx.pl

```diff
@@ -52,13 +55,18 @@
 my $embedDirSourceX64 = "$embedDirRoot/osx-tmp-x86_64";
 my $embedDirSourceARM64 = "$embedDirRoot/osx-tmp-arm64";
 
-print(">>> Building x86_64\n");
-system("perl", "$buildscriptsdir/build.pl", "--clean=1", "--classlibtests=0", "--targetarch=x86_64", @passAlongArgs) eq 0 or die ('failing building x86_64');
 
-if ($artifact)
+# Do not try to build x86_64 for M1 Mac
+if (!$buildMachine && $hostArch != $m1Arch) 
```

Comment:
Nit, I'd take this repeated check and assign it into a named variable like "mergeBinaries" or similar.